### PR TITLE
Chores: remove linter timeout

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind testing

#### What this PR does / why we need it:
Chores: remove linter timeout and use latest golangci-lint
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -30,6 +30,4 @@ jobs:
         uses: golangci/golangci-lint-action@3a919529898de77ec3da873e3063ca4b10e7f5cc # v3.7.0
         with:
           # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
-          version: v1.55
-          args: --timeout 10m -v
-
+          version: latest
```
Comment (lzhecheng) [inline .github/workflows/lint.yaml:33]: I prefer fixed version to avoid breaking change.

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -30,6 +30,4 @@ jobs:
         uses: golangci/golangci-lint-action@3a919529898de77ec3da873e3063ca4b10e7f5cc # v3.7.0
         with:
           # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
-          version: v1.55
-          args: --timeout 10m -v
```
Comment (lzhecheng) [inline .github/workflows/lint.yaml:34]: Shall we keep `-v`?

Comment (MartinForReal) [reply .github/workflows/lint.yaml:34]: fixed.

**Pull request merged**